pkg/apiserver/debugapi/endpoint/endpoint_def.go

```diff
@@ -0,0 +1,685 @@
+	ID:        "pd_region_start_key",
+	Component: model.NodeKindPD,
+	Path:      "/regions/key",
+	Method:    MethodGet,
+	QueryParams: []APIParam{
+		{
+			Name:     "key",
+			Model:    APIParamModelText,
+			Required: true,
+			PreModelTransformer: func(ctx *Context) error {
+				val := ctx.Value()
+				ctx.SetValue(url.QueryEscape(val))
+				return nil
+			},
+		},
+		{
+			Name:  "limit",
+			Model: APIParamModelInt,
+		},
+	},
+}
+
+var pdRegionsStore APIModel = APIModel{
+	ID:        "pd_regions_store",
+	Component: model.NodeKindPD,
+	Path:      "/regions/store/${storeID}",
+	Method:    MethodGet,
+	PathParams: []APIParam{
+		{
+			Name:  "storeID",
+			Model: APIParamModelText,
+		},
+	},
+}
+
+var pdRegionTopRead APIModel = APIModel{
+	ID:        "pd_region_top_read",
+	Component: model.NodeKindPD,
+	Path:      "/regions/readflow",
+	Method:    MethodGet,
+	QueryParams: []APIParam{
+		{
+			Name:  "limit",
+			Model: APIParamModelInt,
+		},
+	},
+}
+
+var pdRegionTopWrite APIModel = APIModel{
+	ID:        "pd_region_top_write",
+	Component: model.NodeKindPD,
+	Path:      "/regions/writeflow",
+	Method:    MethodGet,
+	QueryParams: []APIParam{
+		{
+			Name:  "limit",
+			Model: APIParamModelInt,
+		},
+	},
+}
+
+var pdRegionTopConfVer APIModel = APIModel{
+	ID:        "pd_region_top_conf_ver",
+	Component: model.NodeKindPD,
+	Path:      "/regions/confver",
+	Method:    MethodGet,
+	QueryParams: []APIParam{
+		{
+			Name:  "limit",
+			Model: APIParamModelInt,
+		},
+	},
+}
+
+var pdRegionTopVersion APIModel = APIModel{
+	ID:        "pd_region_top_version",
+	Component: model.NodeKindPD,
+	Path:      "/regions/version",
+	Method:    MethodGet,
+	QueryParams: []APIParam{
+		{
+			Name:  "limit",
+			Model: APIParamModelInt,
+		},
+	},
+}
+
+var pdRegionTopSize APIModel = APIModel{
+	ID:        "pd_region_top_size",
+	Component: model.NodeKindPD,
+	Path:      "/regions/size",
+	Method:    MethodGet,
+	QueryParams: []APIParam{
+		{
+			Name:  "limit",
+			Model: APIParamModelInt,
+		},
+	},
+}
+
+var pdRegionCheck APIModel = APIModel{
+	ID:        "pd_region_check",
+	Component: model.NodeKindPD,
+	Path:      "/regions/check/{state}",
+	Method:    MethodGet,
+	PathParams: []APIParam{
+		{
+			Name:  "state",
+			Model: APIParamModelText,
+		},
+	},
+	QueryParams: []APIParam{
+		{
+			Name:  "bound",
+			Model: APIParamModelInt,
+			PreModelTransformer: func(ctx *Context) error {
+				state := ctx.ParamValue("state")
+				val := ctx.Value()
+
+				if state == "hist-size" {
+					if val != "" {
+						if _, err := strconv.Atoi(val); err != nil {
+							return fmt.Errorf("region size histogram bound should be a number")
+						}
+						ctx.SetValue(val)
+					}
+					ctx.SetValue("10")
+				} else if state == "hist-keys" {
+					if val != "" {
+						if _, err := strconv.Atoi(val); err != nil {
+							return fmt.Errorf("region keys histogram bound should be a number")
+						}
+						ctx.SetValue(val)
+					}
+					ctx.SetValue("10000")
```

Comment:
I'm not sure whether filling these default values at the server side is good :) Since at the browser side users no longer know the request any more.

Reply:
emmm... I've kept the behavior of pd-ctl. Is there any good way to present it in the UI?